[PGO] add extra read/write keys

pianpwk opened the pull request:
Differential Revision: D80321215




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela @mlazos

bobrenjc93 commented on torch/_dynamo/pgo.py:
```diff
@@ -733,9 +736,50 @@ def hit(ty: str) -> defaultdict[CodeId, CodeState]:
+    event_name: Optional[str] = None,
+) -> Optional[defaultdict[CodeId, CodeState]]:
+    code_state = None
+    # TODO: I don't really understand why there's a JSON container format
```
remove?

bobrenjc93 reviewed (approved):
please fix type errors before landing

pianpwk committed:
[PGO] add extra read/write keys (#160715)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/160715

Test Plan:
test_pgo

Rollback Plan:

Reviewed By: bobrenjc93

Differential Revision: D80321215
